src/NoteField.cpp

```diff
@@ -871,38 +869,173 @@ void NoteField::DrawPrimitives()
 
 		float horiz_align= align_right;
 		float side_sign= 1;
-#define draw_all_segments(str_exp, name, caps_name)	\
```

Comment:
I think it's way cleaner to make this into like a templated function or something. Simply inlining looks very bad and would need updating everywhere which is actually less maintainable than this macro.

Reply:
Yeah, I'd also like that a lot better. Based on this and your comment on the other macro changes PR, and the age of this PR (it would cause conflicts in notefield if i update it now) I will make a new one that's better organized and has templates instead of expanded inlines.